### ISSUE TITLE
Request for jail system

### ISSUE THREAD

**Issue by ickycoolboy** [nedhyett/Vermilion#124]
Can there be an option to make it jail a user like ulx does where it jails them where they currently were standing? This is would eliminate the hassle of manually setting jail zones on all the maps my server has and make jailing someone a lot easier.

**Comment by nedhyett** [comment]
You basically mean something that locks a player in place and prevents them from moving/giving input/using tools? If that's the case, then you want "!lockplayer". If not, then would you mind elaborating on your point?

**Comment by ickycoolboy** [comment]
I guess lock player would be what I need. As for jailing though... would it be possible to refine the system more as to set a jail for each map and separate it from the zones part of vermilion? I think that would make it easier to set up the jailing system. Perhaps make the jail part have a completely separate tab for setting it up in the vermilion menu. Then having the !jail command automatically take the jail zone for the current map and send the player to it.

**Comment by nedhyett** [comment]
TL;DR It would be unreliable and time consuming to split the jail system out of the zones module entirely.

Longer version:
The original iteration of the jail system was actually separate from zones. I moved the jail system into the zones system on purpose, as my intention was to allow players to set up regions that players can be jailed in. As far as I understand (I'm not an RPer, so this is just a guess), this would be useful on RP maps, where players are supposed to be locked in a room.

The previous system created an emulated zone that disabled player input if they left a certain radius. This was unreliable and caused a number of unforseen bugs in other addons. Another reason I switched to zones for handling the jail system is that the zone system is very refined in-and-of-itself, whereby it is set up to be as lightweight as possible. I don't see the benefit of splitting them apart again. As well as this, using zones for jails also adds the ability to add special properties to the jail zones, such as weapon confiscation, prop spawn blocking ect.

I do admit, the way in which players are moved into jail is not very obvious and is a bit time consuming. I am planning on making the !jail command move the player to the closest jail to their (or the jailer's) current location if no jail name is specified in the !jail command.

As well as this, while it may be easier for the end user to have a separate jail system, from my perspective, it's much easier to have the jail system part of the zones system as it means less code to maintain (which incidentally means that it's better for the end user because it would be more reliable than an entirely new system, and bugfixes applied to the zones module would also propagate to the jails). The zones system by itself is just under 1.5k lines of code already, so to create a new system, in my opinion, would be bad idea. However, I do intend to move the jail specific stuff to a new module that depends on the zones module to operate.

I'm trying to think of a possible compromise between the two ideas. My current thoughts are:
* Pre-configured zone shapes (i.e. !addcubezone length, !addcuboidzone w h d) to speed up zone creation
* An interface that lists jail zones and the players currently within them (and the ability to take them out or transfer them directly to a new jail)
* Somehow making it more obvious that jailing requires zones

Other than that, I'm also mulling over the idea of having automatic jailing (i.e. based on certain conditions, such as killing someone, votejail, breaking anti-spam rules, spawncamping, receiving too many PP warnings in a set time ect).
